app/controllers/MemberDetailsController.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import cats.implicits.toShow
+import controllers.MemberDetailsController._
+import controllers.actions._
+import forms.NameDOBFormProvider
+import forms.mappings.DateFormErrors
+import models.SchemeId.Srn
+import models.requests.DataRequest
+import models.{DateRange, Mode, NameDOB, NormalMode}
+import navigation.Navigator
+import pages.MemberDetailsPage
+import play.api.data.Form
+import views.html.NameDOBView
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import services.SaveService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import viewmodels.DisplayMessage.SimpleMessage
+import viewmodels.models.NameDOBViewModel
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class MemberDetailsController @Inject()(
+                                         override val messagesApi: MessagesApi,
+                                         navigator: Navigator,
+                                         identifyAndRequireData: IdentifyAndRequireData,
+                                         saveService: SaveService,
+                                         formProvider: NameDOBFormProvider,
+                                         view: NameDOBView,
+                                         val controllerComponents: MessagesControllerComponents
+                                       )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  private val form = MemberDetailsController.form(formProvider)
+
+  def onPageLoad(srn: Srn, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn) {
+    implicit request =>
+      val preparedForm = request.userAnswers.get(MemberDetailsPage(srn)).fold(form)(form.fill)
```

Comment:
There is a method in utils for forms `.fromUserAnswers[A](page: Gettable[A])` that is slightly more readable